### PR TITLE
migrate kubebuilder v3 to v4

### PR DESCRIPTION
**Pull request by Abdiramen**
## What
Moves the ingress controller to ✨ kubebuilder v4 ✨. This way generation to `internal/controller` is handled automagically, and `main.go` is in a more idiomatic location!

original main.go move: https://github.com/ngrok/kubernetes-ingress-controller/pull/14
 kubebuilder's package root move 📦  : https://github.com/kubernetes-sigs/kubebuilder/issues/932
## How
Followed the kubebuilder migration guide: https://book.kubebuilder.io/migration/manually_migration_guide_gov3_to_gov4

## Breaking Changes
Nope!

### CODE REVIEW

**Comment by Abdiramen** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #326** <a href="https://app.graphite.dev/github/pr/ngrok/kubernetes-ingress-controller/326?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ngrok/kubernetes-ingress-controller/326?utm_source=stack-comment).

**Review by alex-bezek — APPROVED** [review]

File: cmd/main.go
Comment (nikolay-ngrok) [inline cmd/main.go:1]: 🤔 should we move this in a subfolder, so default name of this is not `cmd`?

**Review by nikolay-ngrok — APPROVED** [review]

File: cmd/main.go
Comment (Abdiramen) [reply cmd/main.go:1]: We can, but having `main.go` at the top level of `cmd` is a requirement for kubebuilder

Comment (nikolay-ngrok) [reply cmd/main.go:1]: oh nvm then 👍

**Pull request merged**